core/blockchain.go

```diff
@@ -937,11 +948,117 @@ func (bc *BlockChain) WriteBlockWithoutState(block *types.Block, td *big.Int) (e
 	return nil
 }
 
+// GetAccumulatedEpochUptime retrieves the accumulated uptime for an epoch from the database,
+// caching it if found.
+func (bc *BlockChain) GetAccumulatedEpochUptime(epoch uint64) []istanbul.Uptime {
+	uptime := rawdb.ReadAccumulatedEpochUptime(bc.db, epoch)
```

Comment:
Should we acquire a read lock? If not, should this be made non-public? Is this function even necessary?